README.md

```diff
@@ -1,31 +1,52 @@
-# Caramel: A fast read-only key-value store
+# CaramelDB: A High-Performance Succinct Retrieval Library
 
-This repository provides a C++ implementation of the [CARAMEL data structure](https://arxiv.org/pdf/2305.16545.pdf), a space-efficient read-only retrieval table with applications to data management problems in recommender systems, computational biology, and more.
+CaramelDB is a performant C++/Python library of succinct retrieval data structures with applications in data management, machine learning systems, computational biology, and more! The centerpiece of CaramelDB is an implementation of the [CARAMEL data structure](https://arxiv.org/pdf/2305.16545.pdf), a space-efficient table with state-of-the-art performance on practical succinct retrieval workloads. 
 
-## What is CARAMEL?
+## What is Succinct Retrieval?
+
+As defined in [Dillinger, et al. (2022)](https://drops.dagstuhl.de/storage/00lipics/lipics-vol233-sea2022/LIPIcs.SEA.2022.4/LIPIcs.SEA.2022.4.pdf), a *retrieval data structure* $D$ for a static function $f \colon S \to \{0, 1\}^r$ is an index that returns $f(x)$ for any $x \in S$ and any value for $x \notin S$. A highly desirable goal in both theory and practice is to represent $D$ with as little space as possible. To that end, a *succinct retrieval data structure* is an index that uses only $(1 + o(1))r|S|$ bits of space, just slightly more than the information-theoretic lower bound of $r|S|$ bits.
 
-CARAMEL (Compressed Array Representation And More Efficient Lookups) is a data structure built around compressed static functions (CSFs). For a solid introduction to CSFs, see the excellent series of papers by Sebastiano Vigna's group. We draw heavy inspiration from their [Java implementation in Sux4J](https://sux4j.di.unimi.it).
+In the age of massive datasets and machine learning, we observe that numerous applications may benefit from succinct retrieval structures. For example, many industrial search and recommendation systems services reply on computationally expensive machine learning models to compute results. Since executing these queries in real-time is prohibitively expensive, these services often cache the most frequently issues queries along with their corresponding results for faster and more cost-effective serving. As search traffic volume increases over time, it becomes increasingly desirable to store this cache in as few bits as possible via succinct retrieval methods.
```

Comment:
Optional: Can make concise - "In the modern age of massive datasets, numerous applications can benefit from succinct retrieval structures. For example, ... "

Reply:
Done thanks